File: aula_03/src/main/kotlin/com/branas/adapters/domain/Passenger.kt
package com.branas.adapters.domain

import com.branas.adapters.application.repository.entities.PassengerEntity
import java.time.LocalDateTime
import java.util.*

data class Passenger(
    val id: UUID? = null,
    val name: String,
    val email: Email,
    val document: Document,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime? = null
) {
    fun toEntity() = com.branas.adapters.application.repository.entities.PassengerEntity(
        name = name,
        email = email.value,
        document = document.value,
        createdAt = createdAt,
        updatedAt = updatedAt
    )

    companion object {
        fun create(name: String, email: String, document: String, createdAt: LocalDateTime) = Passenger(
            name = name,
            email = Email(email),
            document = Document(document),
            createdAt = createdAt
        )
    }
}

File: aula_06/frontend/src/domain/Passenger.js
import Cpf from "./Cpf.js";
import Email from "./Email.js";
import Name from "./Name.js";

export default class Passenger{
      constructor (passengerId, name, email, document) {
        this.passengerId = passengerId;
        this.name = new Name(name);
        this.email = new Email(email);
        this.document = new Cpf(document);
      }
    
      static create (builder) {
        return new Passenger(builder.passengerId, builder.name, builder.email, builder.document);
      }
}

export class PassengerBuilder {
	constructor (name, email, document) {
    this.name = new Name(name);
    this.email = new Email(email);
    this.document = new Cpf(document);
  }

	build () {
		return Passenger.create(this);
	}
}